### DISCUSSION TITLE
Multiclass classification support

### DISCUSSION THREAD

**Discussion by GillianGrayson** [shenweichen/DeepCTR-Torch#256] [Q&A]
Hi!

Does your package support solving the multiclass classification problem?
I found only one mention for `multiclass` in code:
https://github.com/shenweichen/DeepCTR-Torch/blob/2cd84f305cb50e0fd235c0f0dd5605c8114840a2/deepctr_torch/layers/core.py#L145
I also found only binary cross entropy loss in models

**Comment by joeyzyz** [comment]
I guess it may not support multiclass classification currently, see the code below, it is located [here](https://github.com/shenweichen/DeepCTR-Torch/blob/f6854257b1fc29caab655ce72c6b5528b7432574/deepctr_torch/models/basemodel.py#L252):
`
[loss_func[i](y_pred[:, i], y[:, i], reduction='sum') for i in range(self.num_tasks)])
`
The prediction is hard-coded into a square matrix, thus `y_pred[:, i]` is used to predict binary probability.

**Comment by joeyzyz** [comment]
Also see here, if it is not a binary classification or regression model, it will throw an exception.
https://github.com/shenweichen/DeepCTR-Torch/blob/master/deepctr_torch/models/multitask/mmoe.py#L59
